src/_components/AddAsset/AddAssetContainer.jsx

```diff
@@ -0,0 +1,179 @@
+    if (_.isEmpty(this.props.categories)) {
+      this.props.loadCategories();
+    }
+    if (_.isEmpty(this.props.subcategories)) {
+      this.props.loadSubCategories();
+    }
+    if (_.isEmpty(this.props.assetTypes)) {
+      this.props.loadAssetTypes();
+    }
+    if (_.isEmpty(this.props.assetMakes)) {
+      this.props.loadAssetMakes();
+    }
+    if (_.isEmpty(this.props.modelNumbers)) {
+      this.props.loadModelNumbers();
+    }
+  }
+
+  static getDerivedStateFromProps(nextProps, prevState) {
+    if (nextProps.assets.length !== prevState.assets.length) {
+      ToastMessage.success({ message: 'Asset Saved Successfully' });
+      return {
+        filteredSubCategories: [],
+        filteredAssetTypes: [],
+        filteredAssetMakes: [],
+        filteredModelNumbers: [],
+        modelNumber: 0,
+        assets: nextProps.assets
+      }
+    }
+    return null;
+  }
+
+  handleDropdownChanges = (event, data) => {
+    if(data.name === 'asset-category') {
+      this.setState({
+        filteredSubCategories:
+        filterSubCategories(this.props.subcategories, data.value)
+      });
+    } else if (data.name === 'asset-subcategory') {
+      this.setState({
+        filteredAssetTypes:
+        filterAssetTypes(this.props.assetTypes, data.value)
+      });
+    } else if (data.name === 'asset-types') {
+      this.setState({
+        filteredAssetMakes:
+        filterAssetMakes(this.props.assetMakes, data.value)
+      });
+    } else if (data.name === 'asset-makes') {
+      this.setState({
+        filteredModelNumbers:
+        filterModelNumbers(this.props.modelNumbers, data.value)
+      });
+    }
+  };
+
+  onSelectModelNumber = (event, data) => {
+    this.setState({ modelNumber: data.value });
+  };
+
+  onAddSerialNumber = (event) => {
+    this.setState({ serialNumber: event.target.value });
+  };
+
+  onAddAssetTag = (event) => {
+    this.setState({ assetTag: event.target.value });
+  };
+
+  onCreateAsset = (event, data) => {
+    this.props.createAsset({
+      "asset_code": this.state.assetTag,
+      "serial_number": this.state.serialNumber,
+      "model_number": this.state.modelNumber
+    });
+    event.target.reset();
+  };
+
+  render(){
+    return (
+      <AddAssetComponent {...this.props}
+        handleDropdownChanges={this.handleDropdownChanges}
+        onSelectModelNumber={this.onSelectModelNumber}
+        onAddSerialNumber={this.onAddSerialNumber}
+        onAddAssetTag={this.onAddAssetTag}
+        onCreateAsset={this.onCreateAsset}
+        filteredSubCategories={this.state.filteredSubCategories}
+        filteredAssetTypes={this.state.filteredAssetTypes}
+        filteredAssetMakes={this.state.filteredAssetMakes}
+        filteredModelNumbers={this.state.filteredModelNumbers}
+        modelNumber={this.state.modelNumber}
+        serialNumber={this.state.serialNumber}
+        assetTag={this.state.assetTag}
+      />
+    );
+  }
+}
+
+AddAssetComponent.propTypes = {
```

Comment:
I believe this should be `AddAssetContainer.propTypes`. Also, if there are props that must be present for things to work then add `isRequired` to the validation to make it obvious that the prop is needed in the component